libyul/backends/wasm/BinaryTransform.cpp

```diff
@@ -274,7 +274,7 @@ bytes BinaryTransform::run(Module const& _module)
 
 bytes BinaryTransform::operator()(Literal const& _literal)
 {
-	return toBytes(Opcode::I64Const) + lebEncodeSigned(_literal.value);
+	return toBytes(Opcode::I64Const) + lebEncodeSigned(int64_t(_literal.value));
```

Comment:
Your change is correct. But I wonder how the code works for edge cases. For example, if `_literal.value` is `numeric_limit<int64_t>::max() + 1`.

Reply:
We should use a conversion function that has an assertion.